src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2457,12 +2461,20 @@ def _reindex_axis_0(
             data_column_snowflake_quoted_identifiers.remove(
                 monotonic_increasing_snowflake_quoted_id
             )
+        # When we create an unnamed Series, the default name is set to "__reduced__" in the internal frame.
+        # Don't copy this name to the new frame unless Series has an actual name set.
+        index_column_pandas_labels = (
+            [None]
+            if new_index_modin_frame.data_column_pandas_labels
```

Comment:
once we do the use `to_series` for all index cases then the index labels should be in new_index_modin_frame.index_column_pandas_labels